### ISSUE TITLE
Exit code is zero even when simulator fails

### ISSUE THREAD

**Issue by AugustoPeres** [schism-dev/schism#125]
Hi there,

I have recently started using the schism simulator and noticed that the exit code is zero even when the simulator fails:

```
root@a28729b6320d:/Test_Convergence_Grid1# mpirun --np 3 /schism/build/bin/pschism  
 Must have at least 1 cmd argument: # of scribes to run, or -v for version.
 Must have at least 1 cmd argument: # of scribes to run, or -v for version.
 Must have at least 1 cmd argument: # of scribes to run, or -v for version.
root@a28729b6320d:/Test_Convergence_Grid1# echo $?
0
```

Is there anyway we can have the exit code reflect the fact that the simulation failed?

**Comment by pmav99** [comment]
yeah, we've also had problems with this. As a workaround, we are parsing the stdout and stderr output and have some heuristics that determine if there was an error after all.

For the record, handling this can be even more complicated because the error codes depend on the MPI implementation, too.
For instance, on some tests we did with an older schism version (5.9):
```
openmpi + mpirun -n 8 schism -> error code 0
mpich + mpirun -n 8 schism -> error code 0 or 9 - about 50-50 between them
```

Now this might be an issue with of openmpi/mpich but it could also be an issue of the way schism's MPI code has been implemented. Haven't really looked deeper into it.

**Comment by josephzhang8** [comment]
The error says you need to specify # of scribe processes; see online manual.
